spec/requests/api/v1/users_request_spec.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Users', type: :request do
+  describe 'GET /index' do
+    context 'when the user is not logged in' do
+      before { get api_v1_users_path }
+
+      it 'returns 401 Unauthorized' do
+        expect(response.status).to eq(401)
+      end
+
+      it 'indicates authentication is required' do
+        expect(response.body).to include(CGI.escapeHTML('You need to sign in or sign up before continuing.'))
+      end
+
+      it 'returns a json response' do
+        expect(response.content_type).to eq 'application/json; charset=utf-8'
+      end
+    end
+
+    context 'when the user is logged in' do
+      let!(:users_list) { create_list(:user, 2) }
+      let!(:auth_params) do
+        post api_user_session_path, params: { email: users_list[0].email, password: users_list[0].password }.to_json,
+                                    headers: { 'CONTENT_TYPE' => 'application/json', 'ACCEPT' => 'application/json' }
+        extract_auth_params_from_response(response)
+      end
```

Comment:
Let's just use the simpler way to get the token. You should be able to get one by `user.create_new_auth_token `